### ISSUE TITLE
NotificationReceived and NotificationOpened both trigger

### ISSUE THREAD

**Issue by jkasten2** [OneSignal/OneSignal-Xamarin-SDK#18]
Only `NotificationOpened` should fire when a notification is tapped on. However in the cases below `NotificationReceived` is also firing unexpectedly.

**iOS:**
When an app is completely close and a notification is opened the app launches. On iOS 9 (maybe 10 too). 

**Android:**
When the app is in the background and a notification is opened.

This may only be an issue if the following is set.
```
InFocusDisplaying(Com.OneSignal.OneSignal.OSInFocusDisplayOption.None)
```

**Comment by jamesatgithub** [comment]
hi any update about this bug? it is a block to a key feature in our app. Thanks

**Comment by jkasten2** [comment]
@jamesatgithub We are making some updates to our native SDK that may fix the iOS part of this issue.

Can you include which platform and OS versions you are also seeing this issue on?

**Comment by jamesatgithub** [comment]
tested on iOS 8.0 and 9.0 and 1.0

**Comment by jkasten2** [comment]
@jamesatgithub I just released a 2.1.0 SDK that should fix this issue. Feel free to reopen this issue if you are still seeing a problem.

Thanks.

**Comment by jamesatgithub** [comment]
Hey with new SDK, got below error on iOS simulator. I understand that the push won't work with simulator but the older version doesn't show this error, and the error below failed us to test our app on simulator 

Unhandled Exception:
ObjCRuntime.RuntimeException: Cannot get the method descriptor for the selector 'oneSignalApplicationDidBecomeActive:' on the type 'EDocine.iOS.AppDelegate', because the selector does not correspond to a method
  at (wrapper managed-to-native) UIKit.UIApplication:UIApplicationMain (int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Users/builder/data/lanes/3969/44931ae8/source/xamarin-macios/src/UIKit/UIApplication.cs:79 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x00038] in /Users/builder/data/lanes/3969/44931ae8/source/xamarin-macios/src/UIKit/UIApplication.cs:63
